libbeat/common/transport/tlscommon/tls_config.go

```diff
@@ -151,10 +159,41 @@ func (c *TLSConfig) BuildServerConfig(host string) *tls.Config {
 	return config
 }
 
+func trustESRootCA(cfg *TLSConfig, peerCerts []*x509.Certificate) error {
+	fingerprint, err := hex.DecodeString(cfg.ESCAFingerprint)
+	if err != nil {
+		return fmt.Errorf("decode fingerprint: %w", err)
+	}
+
+	for _, cert := range peerCerts {
+		// Compute digest for each certificate.
+		digest := sha256.Sum256(cert.Raw)
+
+		if bytes.Compare(digest[0:], fingerprint) == 0 {
+			// Make sure the fingerprint matches a CA certificate
+			if cert.IsCA {
+				if cfg.RootCAs == nil {
+					cfg.RootCAs = x509.NewCertPool()
+				}
+				cfg.RootCAs.AddCert(cert)
+
+				cfg.CASha256 = append(cfg.CASha256, Fingerprint(cert))
+			}
+		}
+	}
+
+	return nil
```

Comment:
Should we return an error here?

Reply:
What would be the error? No matching fingerprint?

Reply:
Yes, because this fingerprint is also acting as a pin. I've updated the code to return an error.